src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,44 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int minAge = 18;
+    private static final int minCredentialLength = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can not be null");
+        }
+        Integer userAge = user.getAge();
+        if (userAge == null || userAge < minAge) {
+            String exceptionMessage =
+                    userAge == null ? "Age mustn't be null"
+                            : "You must be 18 years old in order to register";
+            throw new RegistrationException(exceptionMessage);
+        }
+        String userLogin = user.getLogin();
+        if (userLogin == null || userLogin.length() < minCredentialLength) {
+            String exceptionMessage =
+                    userLogin == null ? "Login mustn't be null"
+                            : "Your login must be at least 6 characters long";
```

Comment:
The exception message 'Your login must be at least 6 characters long' is tied to a specific length and may change over time. It's better to use the 'minCredentialLength' constant in the message to avoid hardcoding the length value.